server/src/lib/gist/fetch.ts

```diff
@@ -0,0 +1,72 @@
+import fetch from "node-fetch"
+import { Response } from "node-fetch"
+import { Gist, GistFile } from "./types"
+
+async function fetchHelper(response: Response): Promise<Response> {
+	if (!response.ok) {
+		const isJson = response.headers
+			.get("content-type")
+			?.includes("application/json")
+		const err = await (isJson ? response.json() : response.text())
+		console.warn("Couldn't fetch because", err)
```

Comment:
Do we need to print this on the server? I think its enough to send it (or at least the message) back to the client. Maybe wrap it in a `NODE_ENV === 'development'`?